spanner/api/Spanner.Samples.Tests/CreateInstanceTest.cs

```diff
@@ -29,8 +35,38 @@ public CreateInstanceTest(SpannerFixture spannerFixture)
     public void TestCreateInstance()
     {
         CreateInstanceSample createInstanceSample = new CreateInstanceSample();
-        // Instance already exists since it was created in the test setup so it should throw an exception.
-        var exception = Assert.Throws<RpcException>(() => createInstanceSample.CreateInstance(_spannerFixture.ProjectId, _spannerFixture.InstanceId));
+        RpcException exception;
+        int attempt = 0;
+        while (true)
+        {
+            // Instance already exists since it was created in the test setup so it should throw an exception.
+            exception = Assert.Throws<RpcException>(() => createInstanceSample.CreateInstance(_spannerFixture.ProjectId, _spannerFixture.InstanceId));
+            attempt++;
+            // Retry the test if we get a temporary Unavailable error.
+            if (StatusCode.Unavailable == exception.StatusCode && attempt < MaxAttempts)
+            {
+                var delay = ExtractRetryDelay(exception);
+                Thread.Sleep(delay.Milliseconds);
+                continue;
+            }
+            break;
+        }
         Assert.Equal(StatusCode.AlreadyExists, exception.StatusCode);
```

Comment:
It's fine as it is, but maybe this is cleaner? Just a suggestions though.
```suggestion
        // Instance already exists since it was created in the test setup so it should throw an exception.
        RpcException Attempt() => Assert.Throws<RpcException>(() => createInstanceSample.CreateInstance(_spannerFixture.ProjectId, _spannerFixture.InstanceId));
        
        for (int attempt = 0, RpcException ex = Attempt(); attempt < 10 && ex.StatusCode == StatusCode.Unavailable; attempt++, ex = Attempt())
        {
            var delay = ExtractRetryDelay(exception);
            Thread.Sleep(delay.Milliseconds);
        }
        Assert.Equal(StatusCode.AlreadyExists, exception.StatusCode);
```

Reply:
It's certainly more compact, but I personally find it harder to read and quickly understand what it does.